lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -1664,6 +1665,25 @@ class DecomposeAtenIndexPutHackedTwinOp
 };
 } // namespace
 
+namespace {
+// Decompose `aten.pad` op into `aten.constant_pad_nd` op.
+class DecomposeAtenPadOp : public OpRewritePattern<AtenPadOp> {
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenPadOp op,
+                                PatternRewriter &rewriter) const override {
+
+    Value value = op.value();
```

Comment:
need to guard for OptionalType as well. If it's OptionalType need to report error.

Reply:
Added.